C-Plus-Plus/search/exponential_search.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ Created by Sarthak-9
```

Comment:
Add more description about Exponential search. i.e - How it works and in what cases it should be preferred. Also mention the space and time complexity of the algorithm.